### ISSUE TITLE
Add explicit "use client" to our React components

### ISSUE THREAD

**Issue by mimarz** [digdir/designsystemet#1581]
We are currently adding ["use client"](https://react.dev/reference/react/use-client) in our transpile step to all our files. 

We should add this explicitly to each file instead so that users inspecting the source-code can easily identify its needed for said component code to work with frameworks supporting React Server Components. 

This will also give use the ability to start partially supporting React Server Components for some of our components in the future (by removing "use client")

**Comment by Barsnes** [comment]
Do you want to do this after the pakcage restructuring?

**Comment by Barsnes** [comment]
Will be done after #1570

**Comment by Barsnes** [comment]
After I started working on this task, I noticed Rollup automatically removes directives, such as `'use client';`.
There is an [open issue](https://github.com/rollup/rollup/issues/4699) in Rollup on this issue, and since the `use client` directive is [still in canary](https://react.dev/reference/react/use-client), it might be added as an option later on.

Looked into using a [plugin](https://github.com/Ephem/rollup-plugin-preserve-directives), but since this issue is not a task we need to do now - adding a banner looks like the best solution at the moment.

**Comment by mimarz** [comment]
Should we just close this for now then? @Barsnes

**Comment by Barsnes** [comment]
> Should we just close this for now then? @Barsnes

Since it is not something we can do now, we can close it. If rollup adds this in a later release, we can do it then.
Out current solution works fine for now

**Comment by mimarz** [comment]
Indeed. 

We'll pickup up this once Rollup and/or React gets a new major update.
